### PR TITLE
[BCFR-147][solana] - Add codec chain agnostic modifier for converting byte array address to string

### PR DESCRIPTION
**Pull request by Farber98**
### **Description**
This PR focuses on adding the `AddressBytesToStringModifierConfig` inside chain reader service tests.

### Resources
- [BCFR-147](https://smartcontract-it.atlassian.net/browse/BCFR-147)

### Related PRs
-  https://github.com/smartcontractkit/chainlink-common/pull/818
-  https://github.com/smartcontractkit/chainlink/pull/14620

[BCFR-147]: https://smartcontract-it.atlassian.net/browse/BCFR-147?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: pkg/solana/chainreader/chain_reader_test.go
Comment (ilija42) [inline pkg/solana/chainreader/chain_reader_test.go]: why are we skipping this whole test case, wdym by until we have InputModifications?

**Review by ilija42 — COMMENTED** [review]
Looks good, I just don't understand the skipped test case

File: pkg/solana/codec/byte_string_modifier.go
Comment (ilija42) [inline pkg/solana/codec/byte_string_modifier.go]: comment the why, not the what, a lot of the comments in this file can be very easily derived from the code

File: pkg/solana/chainreader/chain_reader_test.go
Comment (Farber98) [reply pkg/solana/chainreader/chain_reader_test.go]: Was the only way I found to `isolate / skip` this failing test that we talked

I think the same happens in `core` but it is handled with InputModifications:
- [common test](https://github.com/smartcontractkit/chainlink-common/blob/167715aa8613ebdaa17f1b5b4b003a31c120d455/pkg/types/interfacetests/chain_components_interface_tests.go#L272-L273)
- [core config](https://github.com/smartcontractkit/chainlink/blob/develop/core/services/relay/evm/evmtesting/chain_components_interface_tester.go#L185-L191)

I did the same with this new modifier to make that test pass in core:
- [input modification address byte string modifier](https://github.com/smartcontractkit/chainlink/blob/8830519a04ffae6923f5c63130d97fde4a03dd5a/core/services/relay/evm/evmtesting/chain_components_interface_tester.go#L209-L212)

Didn't find a way to achieve the same here, let me know if you think of a better alternative

Comment (ilija42) [reply pkg/solana/chainreader/chain_reader_test.go]: We shouldn't break unrelated tests like this. Can the struct be manually "retyped" with an added TODO when this test case is encountered instead of using an input modifier

File: pkg/solana/chainreader/chain_reader_test.go
Comment (ilija42) [inline pkg/solana/chainreader/chain_reader_test.go]: The test is still being skipped, just delete this Skip and everything should work
```suggestion
```

**Review by ilija42 — CHANGES_REQUESTED** [review]

**Review by ilija42 — DISMISSED** [review]

**Review by silaslenihan — DISMISSED** [review]

File: pkg/solana/chainreader/chain_reader.go
Comment (jmank88) [inline pkg/solana/chainreader/chain_reader.go]: Will this be used from anywhere else? IMHO it would read easier inline. Otherwise, this func could accept just the `OutputModifications` instead of a reference to the enclosing `ChainReaderProcedure`

Comment (EasterTheBunny) [reply pkg/solana/chainreader/chain_reader.go]: Well the init function is already 3 deep on nested for loops. This just adds another. Layering like that gets hard to read inline too. Kind of a style issue. Linters would normalize this though.

**Review by EasterTheBunny — DISMISSED** [review]

File: pkg/solana/chainreader/chain_reader.go
Comment (jmank88) [reply pkg/solana/chainreader/chain_reader.go]: Agree with limiting nesting, but this is not the only way. IMHO outside-in would yield cleaner results. How about using a helper for everything after:
```go
for namespace, methods := range namespaces {
		for methodName, method := range methods.Methods {
```

Comment (jmank88) [reply pkg/solana/chainreader/chain_reader.go]: If not, then please simplify this func to accept only what is necessary, and find a clearer name

**Review by ilija42 — DISMISSED** [review]

**Review by silaslenihan — DISMISSED** [review]

File: pkg/solana/chainreader/chain_reader_test.go
Comment (jmank88) [inline pkg/solana/chainreader/chain_reader_test.go:685]: Do we have a ticket to link?

**Review by jmank88 — DISMISSED** [review]

**Review by aalu1418 — DISMISSED** [review]
lgtm! i will enable auto-merge once the common PR is merged and ref updated

**Review by ilija42 — APPROVED** [review]

**Review by aalu1418 — APPROVED** [review]

**Pull request merged**
